src/clj_kondo/impl/analyzer/re_frame.clj

```diff
@@ -40,7 +50,46 @@
           (common/analyze-children ctx args)))
       (common/analyze-children ctx children))))
 
-(defn analyze-reg-sub [ctx expr fq-def]
+(defn analyze-inject-cofx
+  "Analyzes re-frame co effect injection.
+  Marks the co effect id, the first parameter as a co effect reference so it can be tracked back
+  to the co effect registration. Marker is `[:context :re-frame.core :cofx-ref]`.
+  https://day8.github.io/re-frame/api-re-frame.core/#inject-cofx"
+  [ctx expr ns]
+  (let [kns (keyword ns)
+        [cofx-id & cofx-params] (next (:children expr))]
+    (common/analyze-children (assoc-in ctx [:context kns :cofx-ref] true) [cofx-id])
+    (when cofx-params
+      (common/analyze-children ctx cofx-params))))
+
+(defn- analyze-dispatch-event-id
+  "Marks the event id, the first element of the parameter vector as an event reference so
+  it can be tracked back to the event registration. Marker is `[:context :re-frame.core :event-ref]`.
+  https://day8.github.io/re-frame/api-re-frame.core/#dispatch"
+  [ctx expr ns]
+  (let [kns (keyword ns)
+        [farg & _args :as _children] (next (:children expr))]
+    (when (identical? :vector (utils/tag farg))
```

Comment:
@benedekfazekas What if the argument is not a vector, then it won't be analyzed? E.g.

```
(let [x ...]
  (dispatch x))
```

Reply:
essentially yes, this is a known limitation -- I think there was an issue about this as well and we discussed it a bit maybe on slack?! -- not sure. thinking about it but I don't think this is the scope if this PR. unless of course you think that without taking care of these cases this causes more harm than good

Reply:
I'm not talking about correct analysis, my concern is when skipping analysis for some parts, that will cause false positive lint warnings like "unused binding".

Reply:
This seems to work:
```
clojure -M:clj-kondo/dev --lint - <<< '(ns foo (:require [re-frame.core :as r])) (let [x nil] (r/dispatch x))'
linting took 34ms, errors: 0, warnings: 0
```

but why does that work?

Reply:
because this fn just analyzes the event-id as the name shows. and this case it does not even analyzes that because it only understands the default format, i.e. a vector. the entry point for a direct dispatch call however -- see `analyze-dispatch` --, analyzes the whole form. does this make more sense?

Reply:
Makes sense